packages/botonic-nlp/src/tasks/ner/botonic-ner.ts

```diff
@@ -21,40 +22,44 @@ import {
 } from './process'
 import { NerConfig, NerConfigStorage } from './storage'
 
+export interface EntityRecognizerConfig extends NlpTaskConfig {
+  entities: string[]
+}
+
 export class BotonicNer {
+  readonly locale: Locale
+  readonly maxLength: number
+  readonly vocabulary: string[]
   readonly entities: string[]
   private processor: Processor
   private predictionProcessor: PredictionProcessor
   private modelManager: ModelManager
 
-  constructor(
-    readonly locale: Locale,
-    readonly maxLength: number,
-    entities: string[],
-    readonly vocabulary: string[],
-    private readonly preprocessor: Preprocessor
-  ) {
-    this.entities = unique([NEUTRAL_ENTITY].concat(entities))
+  constructor(config: EntityRecognizerConfig) {
+    this.locale = config.locale
```

Comment:
I don't that BotonicNer and BotonicIntentClassifier have so many fields. They are difficult to maintain & document, and since you use them in 2 different places, encapsulating them you would be worth.

Can you try this?
```
export class BotonicNer {
   readonly (private if possible) config: Readonly<EntityRecognizerConfig>
   constructor(config: EntityRecognizerConfig) {
       this.config = {...config, entities : unique([NEUTRAL_ENTITY].concat(config.entities))}
   }
}
```

Reply:
I totally agree with you, I was going to apply something similar to this in the next PR. 🤔